src/interpreter/flow.rs

```diff
@@ -86,8 +86,8 @@ where
             return Err(PanicReason::ContractNotInInputs.into());
         }
 
-        // TODO validate external and internal context
-        // TODO update asset ID balance
+        // credit contract asset_id balance
```

Comment:
Need to add the internal context part:
https://github.com/FuelLabs/fuel-specs/blob/a65e14dfe5c0049ebf4f8e0b33b061c60927d7d9/specs/vm/opcodes.md?plain=1#L1091

Reply:
done and unit tested